arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ArrowReader.java

```diff
@@ -206,15 +238,28 @@ public void close() throws IOException {
         boolean caseSensitive,
         int batchSize,
         boolean reuseContainers) {
-      this.fileTasks = StreamSupport.stream(tasks.spliterator(), false)
+      List<FileScanTask> fileTasks = StreamSupport.stream(tasks.spliterator(), false)
           .map(CombinedScanTask::files)
           .flatMap(Collection::stream)
           .collect(Collectors.toList());
       this.fileItr = fileTasks.iterator();
 
+      if (fileTasks.stream().anyMatch(TableScanUtil::hasDeletes)) {
+        throw new UnsupportedOperationException("Cannot read files that require applying delete files");
+      }
+      if (expectedSchema.columns().isEmpty()) {
```

Comment:
Style nit: Iceberg adds an empty newline after control flow statements like `if`/`else`/`for`/`while`.

Reply:
fixed